session_0/word_processing.rb

```diff
@@ -34,11 +38,31 @@ def words_with_prefix(prefix, words)
 # The function `similarity_score` takes two words and returns the
 # similarity score (an integer).
 def similarity_score(word_1, word_2)
-  raise NotImplementedError # TODO
+  d=0
+  for i in 1..(word_1.length<word_2.length ? word_1.length: word_2.length)
+    if (word_1[i-1]==word_2[i-1])
```

Comment:
try using slices, it improves readability of the code
https://apidock.com/ruby/String/slice

You can also avoid the ternary operator by using max method
https://stackoverflow.com/questions/1359370/how-to-find-a-min-max-with-ruby